Local test failures

Tests should pass when run with `go test`, with or without `GOLANG_SAMPLES_PROJECT_ID`.

Some of these should legit fail due to API not being enabled on the test project, etc. but they have confusing test output, including some that panic, leaving a large stack trace to dig through.

```
$ GOLANG_SAMPLES_PROJECT_ID=cbro-scratch go test -v ./...

<snip>
```

asset:

```
=== RUN   TestMain
--- FAIL: TestMain (34.26s)
    main_test.go:67: execution failed: exit status 1
    main_test.go:70: did not expect stderr output, got 216 bytes: 2020/05/13 22:19:55 client.CreateFeed: rpc error: code = InvalidArgument desc = Fail to use projects/cbro-scratch/topics/YOUR_TOPIC_NAME as feed output destination. Error: User not authorized to perform this action.
    main_test.go:74: stdout returned , wanted to contain FEED-1589408361145568802
FAIL
FAIL    github.com/GoogleCloudPlatform/golang-samples/asset/quickstart/create-feed      34.772s
=== RUN   TestMain
--- FAIL: TestMain (2.30s)
    main_test.go:65: client.CreateFeed: rpc error: code = InvalidArgument desc = Fail to use projects/cbro-scratch/topics/YOUR_TOPIC_NAME as feed output destination. Error: User not authorized to perform this action.
FAIL
FAIL    github.com/GoogleCloudPlatform/golang-samples/asset/quickstart/delete-feed      2.799s
=== RUN   TestMain
2020/05/13 22:20:02 rpc error: code = PermissionDenied desc = service-618104708054@gcp-sa-cloudasset.iam.gserviceaccount.com does not have storage.objects.create access to cbro-scratch-for-assets/my-assets.txt.
FAIL    github.com/GoogleCloudPlatform/golang-samples/asset/quickstart/export-assets    3.517s
=== RUN   TestMain
--- FAIL: TestMain (2.22s)
    main_test.go:76: client.CreateFeed: rpc error: code = InvalidArgument desc = Fail to use projects/cbro-scratch/topics/YOUR_TOPIC_NAME as feed output destination. Error: User not authorized to perform this action.
FAIL
FAIL    github.com/GoogleCloudPlatform/golang-samples/asset/quickstart/get-feed 2.254s
=== RUN   TestMain
--- FAIL: TestMain (1.80s)
    main_test.go:76: client.CreateFeed: rpc error: code = InvalidArgument desc = Fail to use projects/cbro-scratch/topics/YOUR_TOPIC_NAME as feed output destination. Error: User not authorized to perform this action.
FAIL
FAIL    github.com/GoogleCloudPlatform/golang-samples/asset/quickstart/list-feeds       2.025s
=== RUN   TestMain
--- FAIL: TestMain (3.28s)
    main_test.go:80: client.CreateFeed: rpc error: code = InvalidArgument desc = Fail to use projects/cbro-scratch/topics/YOUR_TOPIC_NAME as feed output destination. Error: User not authorized to perform this action.
FAIL
FAIL    github.com/GoogleCloudPlatform/golang-samples/asset/quickstart/update-feed      3.490s
```

bigquery cmek:
```
--- FAIL: TestTables (13.36s)
    --- FAIL: TestTables/cmektests (0.34s)
        integration_test.go:122: createTableWithCMEK("snippet_table_tests_b1c66b52f6dd4e62ad3c0fbbdcfe47b2" "testtable_9b2fa3498a4143e498ef363c9c2e26b8"): googleapi: Error 403: Access Denied: BigQuery BigQuery: Cloud KMS Error: Permission 'cloudkms.cryptoKeyVersions.useToEncrypt' denied on resource 'projects/cloud-samples-tests/locations/us/keyRings/test/cryptoKeys/test' (or it may not exist). Please grant Cloud KMS CryptoKey Encrypter/Decrypter role to BigQuery service account: bq-534326803979@bigquery-encryption.iam.gserviceaccount.com, accessDenied
FAIL
FAIL    github.com/GoogleCloudPlatform/golang-samples/bigquery/snippets/table   13.371s
```

dataproc

> cbro: this should probably be changed to an E2E test

```
=== RUN   TestCreateCluster
--- PASS: TestCreateCluster (358.99s)
=== RUN   TestInstantiateInlineWorkflowTemplate
panic: test timed out after 10m0s
goroutine 131 [running]:
testing.(*M).startAlarm.func1()
        /usr/local/go/src/testing/testing.go:1377 +0xdf
created by time.goFunc
        /usr/local/go/src/time/sleep.go:168 +0x44
goroutine 1 [chan receive, 5 minutes]:
testing.(*T).Run(0xc0002d8f00, 0xa6c9b5, 0x25, 0xa83500, 0x489201)
        /usr/local/go/src/testing/testing.go:961 +0x377
testing.runTests.func1(0xc0000fc100)
        /usr/local/go/src/testing/testing.go:1202 +0x78
testing.tRunner(0xc0000fc100, 0xc000241dc0)
        /usr/local/go/src/testing/testing.go:909 +0xc9
testing.runTests(0xc0000b6f40, 0xf15640, 0x2, 0x2, 0x0)
        /usr/local/go/src/testing/testing.go:1200 +0x2a7
testing.(*M).Run(0xc000106200, 0x0)
        /usr/local/go/src/testing/testing.go:1117 +0x176
main.main()
        _testmain.go:46 +0x135
goroutine 20 [select]:
<snip>
        /home/cbro/gopath/pkg/mod/golang.org/x/net@v0.0.0-20200324143707-d3edc9973b7e/http2/frame.go:237 +0x87
golang.org/x/net/http2.(*Framer).ReadFrame(0xc000324380, 0xc000318390, 0xc000318300, 0x0, 0x0)
        /home/cbro/gopath/pkg/mod/golang.org/x/net@v0.0.0-20200324143707-d3edc9973b7e/http2/frame.go:492 +0xa1
google.golang.org/grpc/internal/transport.(*http2Client).reader(0xc000479dc0)
        /home/cbro/gopath/pkg/mod/google.golang.org/grpc@v1.28.0/internal/transport/http2_client.go:1270 +0x183
created by google.golang.org/grpc/internal/transport.newHTTP2Client
        /home/cbro/gopath/pkg/mod/google.golang.org/grpc@v1.28.0/internal/transport/http2_client.go:300 +0xd23

goroutine 106 [select, 3 minutes]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc0002dba90, 0x1, 0x0, 0x0, 0x0, 0x0)
        /home/cbro/gopath/pkg/mod/google.golang.org/grpc@v1.28.0/internal/transport/controlbuf.go:395 +0x122
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc0002087e0, 0x0, 0x0)
        /home/cbro/gopath/pkg/mod/google.golang.org/grpc@v1.28.0/internal/transport/controlbuf.go:513 +0x1e3
google.golang.org/grpc/internal/transport.newHTTP2Client.func3(0xc000479880)
        /home/cbro/gopath/pkg/mod/google.golang.org/grpc@v1.28.0/internal/transport/http2_client.go:346 +0x7b
created by google.golang.org/grpc/internal/transport.newHTTP2Client
        /home/cbro/gopath/pkg/mod/google.golang.org/grpc@v1.28.0/internal/transport/http2_client.go:344 +0xedb
FAIL    github.com/GoogleCloudPlatform/golang-samples/dataproc  600.014s
=== RUN   TestQuickstart
```

dlp:
```
=== RUN   TestListJobs
--- FAIL: TestListJobs (8.16s)
    jobs_test.go:159: "Job" not found in listJobs output: ""
=== RUN   TestDeleteJob
--- FAIL: TestDeleteJob (3.95s)
panic: runtime error: index out of range [1] with length 0 [recovered]
        panic: runtime error: index out of range [1] with length 0

goroutine 134 [running]:
testing.tRunner.func1(0xc0000f8f00)
        /usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0xa8b300, 0xc0001b3ca0)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/GoogleCloudPlatform/golang-samples/dlp/snippets/jobs.TestDeleteJob(0xc0000f8f00)
        /home/cbro/golang-samples/dlp/snippets/jobs/jobs_test.go:177 +0x4e1
testing.tRunner(0xc0000f8f00, 0xb1fb90)
        /usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:960 +0x350
FAIL    github.com/GoogleCloudPlatform/golang-samples/dlp/snippets/jobs 12.123s
--- FAIL: TestRisk (0.00s)
    --- FAIL: TestRisk/Numerical (59.75s)
        retry.go:44: FAILED after 20 attempts:
            risk_test.go:100: riskKMap got , want substring "Created job"
    --- FAIL: TestRisk/L_Diversity (62.16s)
        retry.go:44: FAILED after 20 attempts:
            risk_test.go:100: riskKMap got , want substring "Created job"
    --- FAIL: TestRisk/K_Map (57.54s)
        retry.go:44: FAILED after 20 attempts:
            risk_test.go:100: riskKMap got , want substring "Created job"
    --- FAIL: TestRisk/K_Anonymity (57.76s)
        retry.go:44: FAILED after 20 attempts:
            risk_test.go:100: riskKMap got , want substring "Created job"
    --- FAIL: TestRisk/Categorical (59.51s)
        retry.go:44: FAILED after 20 attempts:
            risk_test.go:100: riskKMap got , want substring "Created job"
FAIL
FAIL    github.com/GoogleCloudPlatform/golang-samples/dlp/snippets/risk 176.819s
=== RUN   TestTriggersSamples
--- FAIL: TestTriggersSamples (0.69s)
    trigger_test.go:47: createTrigger: CreateJobTrigger: rpc error: code = PermissionDenied desc = Assure that the DLP API is enabled and billing is turned on in the project referenced by the parent field. Cloud DLP API must be enabled in 534326803979 so that the DLP API service account service-534326803979@dlp-api.iam.gserviceaccount.com is generated for accessing GCP resources being scanned or used by DLP. Project has the service disabled.
    trigger_test.go:50: createTrigger got
        ----

        ----
        Want to contain:
        ----
        Successfully created trigger
        ----
    trigger_test.go:58: listTriggers got
        ----

        ----
        Want to contain:
        ----
        projects/cbro-scratch/jobTriggers/my-trigger
        ----
    trigger_test.go:63: deleteTrigger: DeleteJobTrigger: rpc error: code = NotFound desc = Unknown trigger "projects/cbro-scratch/jobTriggers/my-trigger".
    trigger_test.go:66: deleteTrigger got
        ----

        ----
        Want to contain:
        ----
        Successfully deleted trigger
        ----
FAIL
FAIL    github.com/GoogleCloudPlatform/golang-samples/dlp/snippets/trigger      0.698s
```

healthcare:
```
=== RUN   TestFHIRStore
--- FAIL: TestFHIRStore (118.39s)
    retry.go:44: FAILED after 10 attempts:
        fhir_test.go:262: exportFHIRResource got err: Export: googleapi: Error 403: Cloud Healthcare Service Agent doesn't have all required permissions [storage.objects.create storage.objects.delete storage.objects.list] on resource "gs://cbro-scratch-healthcare-test" (or the resource may not exist), forbidden
        fhir_test.go:266: importFHIRResource got err: Import: googleapi: Error 403: Permission 'storage.objects.list' denied on resource 'gs://cbro-scratch-healthcare-test' (or it may not exist)., forbidden
=== RUN   TestHL7V2Store
--- PASS: TestHL7V2Store (1.06s)
FAIL
FAIL    github.com/GoogleCloudPlatform/golang-samples/healthcare        203.047s
```

kms:
```
=== RUN   TestCreateKeyRing
--- SKIP: TestCreateKeyRing (0.00s)
    snippets_test.go:134: TestCreateKeyRing skipped. There's currently no method to delete keyrings, so we should avoid creating resources
=== RUN   TestCreateCryptoKey
--- SKIP: TestCreateCryptoKey (0.00s)
    snippets_test.go:154: TestCreateCryptoKey skipped. There's currently no method to delete keys, so we should avoid creating resources
=== RUN   TestChangeKeyVersionState
--- FAIL: TestChangeKeyVersionState (1.93s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x8db70d]

goroutine 407 [running]:
testing.tRunner.func1(0xc000716300)
        /usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x97ba40, 0xed9380)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/GoogleCloudPlatform/golang-samples/kms.TestChangeKeyVersionState.func1(0xc000421240)
        /home/cbro/golang-samples/kms/snippets_test.go:192 +0x11d
github.com/GoogleCloudPlatform/golang-samples/internal/testutil.Retry(0xc000716300, 0x14, 0x12a05f200, 0xc000147de0, 0x1)
        /home/cbro/golang-samples/internal/testutil/retry.go:34 +0xc3
github.com/GoogleCloudPlatform/golang-samples/kms.TestChangeKeyVersionState(0xc000716300)
        /home/cbro/golang-samples/kms/snippets_test.go:184 +0x1fe
testing.tRunner(0xc000716300, 0xa6b830)
        /usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:960 +0x350
FAIL    github.com/GoogleCloudPlatform/golang-samples/kms       36.523s
```

speech:
```
=== RUN   TestRecognize
--- FAIL: TestRecognize (0.12s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x8c30d0]

goroutine 21 [running]:
testing.tRunner.func1(0xc0000ea200)
        /usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x95fd60, 0xea30e0)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/GoogleCloudPlatform/golang-samples/speech/caption.recognize(0xada7e0, 0xc000115530, 0xa28b78, 0x14, 0x19, 0xbf0420)
        /home/cbro/golang-samples/speech/caption/caption.go:126 +0x210
github.com/GoogleCloudPlatform/golang-samples/speech/caption.TestRecognize(0xc0000ea200)
        /home/cbro/golang-samples/speech/caption/caption_test.go:30 +0x7f
testing.tRunner(0xc0000ea200, 0xa499b0)
        /usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:960 +0x350
FAIL    github.com/GoogleCloudPlatform/golang-samples/speech/caption    0.130s
=== RUN   TestRecognize
--- FAIL: TestRecognize (0.14s)
```

storage:
```
=== RUN   TestV4SignedURL
--- FAIL: TestV4SignedURL (0.56s)
    objects_test.go:218: generateV4PutObjectSignedURL: ioutil.ReadFile: open : no such file or directory
    objects_test.go:222: got "", want "Generated PUT signed URL:"
    objects_test.go:231: httpClient.Do: Put : unsupported protocol scheme ""
    objects_test.go:236: generateV4GetObjectSignedURL: ioutil.ReadFile: open : no such file or directory
    objects_test.go:240: got "", want "Generated GET signed URL:"
    objects_test.go:245: http.Get: Get : unsupported protocol scheme ""
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x8fc510]

goroutine 214 [running]:
testing.tRunner.func1(0xc0001db200)
        /usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x9a3680, 0xee3f90)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/GoogleCloudPlatform/golang-samples/storage/objects.TestV4SignedURL(0xc0001db200)
        /home/cbro/golang-samples/storage/objects/objects_test.go:247 +0x880
testing.tRunner(0xc0001db200, 0xa8b260)
        /usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:960 +0x350
FAIL    github.com/GoogleCloudPlatform/golang-samples/storage/objects   7.608s
```

tasks:
```
--- FAIL: TestCreateHTTPTask (0.24s)
    create_http_task_test.go:46: CreateTask(Message): cloudtasks.CreateTask: rpc error: code = FailedPrecondition desc = Queue does not exist.
    create_http_task_test.go:46: CreateTask(No Message): cloudtasks.CreateTask: rpc error: code = FailedPrecondition desc = Queue does not exist.
FAIL
FAIL    github.com/GoogleCloudPlatform/golang-samples/tasks     0.251s
```

--- a/dataproc/instantiate_inline_workflow_template_test.go
+++ b/dataproc/instantiate_inline_workflow_template_test.go
@@ -24,7 +24,7 @@ import (
 )
 
 func TestInstantiateInlineWorkflowTemplate(t *testing.T) {
-	tc := testutil.SystemTest(t)
+	tc := testutil.EndToEndTest(t)
 
 	region := "us-central1"
 

--- a/dlp/snippets/trigger/trigger_test.go
+++ b/dlp/snippets/trigger/trigger_test.go
@@ -44,7 +44,7 @@ func TestTriggersSamples(t *testing.T) {
 	}
 
 	if err := createTrigger(buf, tc.ProjectID, "my-trigger", "My Trigger", "Test trigger", "my-bucket", nil); err != nil {
-		t.Errorf("createTrigger: %v", err)
+		t.Fatalf("createTrigger: %v", err)
 	}
 	if got, want := buf.String(), "Successfully created trigger"; !strings.Contains(got, want) {
 		t.Errorf("createTrigger got\n----\n%v\n----\nWant to contain:\n----\n%v\n----", got, want)

--- a/storage/objects/objects_test.go
+++ b/storage/objects/objects_test.go
@@ -282,6 +282,9 @@ func TestV4SignedURL(t *testing.T) {
 	bucketName := tc.ProjectID + "-signed-url-bucket-name"
 	objectName := "foo.txt"
 	serviceAccount := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+	if serviceAccount == "" {
+		t.Skip("GOOGLE_APPLICATION_CREDENTIALS must be set")
+	}
 
 	testutil.CleanBucket(ctx, t, tc.ProjectID, bucketName)
 	putBuf := new(bytes.Buffer)

